### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #70 

## Description
TBD

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels 

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling for script execution to better capture and manage specific errors.

### CODE REVIEW

File: models/transaction.go [hunk 1]
```diff
@@ -190,24 +190,24 @@ func decodeTransactionEvent(
...
 	encodedTx, err := hex.DecodeString(txEvent.Payload)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to hex-decode transaction payload: %w", err)
+		return nil, nil, fmt.Errorf("failed to hex decode transaction payload [%s]: %w", txEvent.Payload, err)
 	}
 
 	encodedLogs, err := hex.DecodeString(txEvent.Logs)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to hex decode receipt: %w", err)
+		return nil, nil, fmt.Errorf("failed to hex decode receipt [%s]: %w", txEvent.Logs, err)
```
Comment (m-Peter) [inline models/transaction.go:203]:
```suggestion
		return nil, nil, fmt.Errorf("failed to hex decode transaction logs [%s]: %w", txEvent.Logs, err)
```

File: models/transaction.go [hunk 2]
```diff
@@ -190,24 +190,24 @@ func decodeTransactionEvent(
...
 	}
 
 	var logs []*gethTypes.Log
 	if len(encodedLogs) > 0 {
 		err = rlp.Decode(bytes.NewReader(encodedLogs), &logs)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to RLP-decode receipt: %w", err)
+			return nil, nil, fmt.Errorf("failed to RLP-decode receipt [%x]: %w", encodedLogs, err)
```
Comment (m-Peter) [inline models/transaction.go:210]:
```suggestion
			return nil, nil, fmt.Errorf("failed to RLP-decode transaction logs [%x]: %w", encodedLogs, err)
```

File: models/transaction.go [hunk 3]
```diff
@@ -246,15 +246,15 @@ func decodeTransactionEvent(
 	if txEvent.TransactionType == types.DirectCallTxType {
 		directCall, err := types.DirectCallFromEncoded(encodedTx)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to RLP-decode direct call: %w", err)
+			return nil, nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", encodedTx, err)
```
Comment (m-Peter) [inline models/transaction.go:249]: It occurred to me that we have the same exact error message in `decodeTransactionEvent` & `UnmarshalTransaction`:

In `decodeTransactionEvent`:
```go
directCall, err := types.DirectCallFromEncoded(encodedTx)
if err != nil {
	return nil, nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", encodedTx, err)
}

gethTx := &gethTypes.Transaction{}
if err := gethTx.UnmarshalBinary(encodedTx); err != nil {
	return nil, nil, fmt.Errorf("failed to RLP-decode transaction [%x]: %w", encodedTx, err)
}
```

In `UnmarshalTransaction`:
```go
directCall, err := types.DirectCallFromEncoded(value)
if err != nil {
	return nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", value, err)
}

if err := tx.UnmarshalBinary(value); err == nil {
	return TransactionCall{Transaction: tx}, nil
}

return nil, fmt.Errorf("failed to RLP-decode transaction [%x]: %w", value, err)
```

Can we add a prefix to the error message, so we can know whether the decoding error comes the ingestion component, or from the DB?

Comment (sideninja) [reply models/transaction.go:249]: no need because the error in the API gets wrapped.

**Review by m-Peter — APPROVED** [review]
LGTM! :tada:

**Pull request merged**